miden-tx/tests/test_miden_faucet_contract.rs

```diff
@@ -0,0 +1,437 @@
+
+#[derive(Clone)]
+pub enum DataStoreError {
+    AccountNotFound(AccountId),
+    NoteNotFound(u32, NodeIndex),
+}
+
+#[derive(Clone)]
+
+pub struct MockDataStore {
+    pub account: Account,
+    pub block_header: BlockHeader,
+    pub block_chain: ChainMmr,
+    pub notes: Vec<Note>,
+}
+
+impl MockDataStore {
+    pub fn with_existing(account: Option<Account>, consumed_notes: Option<Vec<Note>>) -> Self {
+        let (account, block_header, block_chain, consumed_notes) = mock_inputs_with_existing(
+            MockAccountType::StandardExisting,
+            AssetPreservationStatus::Preserved,
+            account,
+            consumed_notes,
+        );
+        Self {
+            account,
+            block_header,
+            block_chain,
+            notes: consumed_notes,
+        }
+    }
+}
+
+impl DataStore for MockDataStore {
+    fn get_transaction_data(
+        &self,
+        account_id: AccountId,
+        block_num: u32,
+        notes: &[NoteOrigin],
+    ) -> Result<(Account, BlockHeader, ChainMmr, Vec<Note>), miden_tx::DataStoreError> {
+        assert_eq!(account_id, self.account.id());
+        assert_eq!(block_num as u64, self.block_header.block_num().as_int());
+        assert_eq!(notes.len(), self.notes.len());
+        let origins = self
+            .notes
+            .iter()
+            .map(|note| note.proof().as_ref().unwrap().origin())
+            .collect::<Vec<_>>();
+        notes.iter().all(|note| origins.contains(&note));
+        Ok((
+            self.account.clone(),
+            self.block_header.clone(),
+            self.block_chain.clone(),
+            self.notes.clone(),
+        ))
+    }
+
+    fn get_account_code(
+        &self,
+        account_id: AccountId,
+    ) -> Result<ModuleAst, miden_tx::DataStoreError> {
+        assert_eq!(account_id, self.account.id());
+        Ok(self.account.code().module().clone())
+    }
+}
+
+fn get_faucet_account_with_max_suppply(max_supply: u64) -> Account {
+    let faucet_account_id = AccountId::try_from(ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN).unwrap();
+    let faucet_account_code_src = include_str!("../../miden-lib/asm/faucets/basic.masm");
+    let faucet_account_code_ast = ModuleAst::parse(faucet_account_code_src).unwrap();
+    let mut account_assembler = assembler();
+
+    let faucet_account_code = AccountCode::new(
+        faucet_account_id,
+        faucet_account_code_ast.clone(),
+        &mut account_assembler,
+    )
+    .unwrap();
+
+    let mut faucet_account_storage = mock_account_storage();
+    let faucet_storage_slot_1 =
+        [Felt::new(max_supply), Felt::new(0), Felt::new(0), Felt::new(0)];
+    faucet_account_storage.set_item(1, faucet_storage_slot_1);
+
+    Account::new(
+        faucet_account_id,
+        AccountVault::new(&vec![]).unwrap(),
+        faucet_account_storage.clone(),
+        faucet_account_code.clone(),
+        Felt::new(1),
+    )
+}
+
+#[test]
+fn test_faucet_contract_mint_fungible_asset_succeeds() {
```

Comment:
@hackaugusto can you try to run this test here? 

Somehow the created note does not have the recipient that I have defined line 134.